batch-setup/make_meta_tiles.py

```diff
@@ -187,31 +187,50 @@ def missing_tiles_split(self, group_by_zoom, queue_zoom, big_jobs,
             # queue_zoom is always less than or equal to group_by_zoom?
             missing_high = CoordSet(min_zoom=queue_zoom, max_zoom=group_by_zoom)
 
-            coords = self.generate_missing_tile_coords(try_generator)
-            for c in coords:
-                if c.zoom < group_by_zoom:  # 10
+
+            coords_or_file = self.generate_missing_tile_coords(tmpdir, try_generator)
+
+            for c in coords_or_file:
+                if isinstance(coords_or_file, io.IOBase):
```

Comment:
When would this be a file? It's odd to have this check here in a loop that will get called a bunch of times.

Reply:
yep; why it looks like is is that originally, before we introduce customized tiles generator it was a miss_file here, and for that it needs to deserialize; and I had a hard time converge the file and customer generator to have a same interface here, so I gave up and have this isinstance condition here. I think the refactor suggestion Travis pointed out below can improve the readability.